flink-connector-aws/flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/source/proxy/DynamoDbStreamsProxy.java

```diff
@@ -168,8 +194,15 @@ private String getShardIterator(
         try {
             return dynamoDbStreamsClient.getShardIterator(requestBuilder.build()).shardIterator();
         } catch (ResourceNotFoundException e) {
-            LOG.info(
-                    "Received ResourceNotFoundException. "
+            LOG.warn(
+                    "Received ResourceNotFoundException from GetShardIterator. "
+                            + "Shard {} of stream {} is no longer valid, marking it as complete.",
```

Comment:
this is good to be warn. Good call.

Let's add a warning here too that this might indicate data loss (since shard was not fully read, but it already expired)

Reply:
added a warning in the log